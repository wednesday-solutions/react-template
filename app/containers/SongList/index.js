/**
 *
 * SongList
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { createStructuredSelector } from 'reselect';
import { connect } from 'react-redux';

// import styled from 'styled-components'
import { Card, Skeleton } from 'antd';
import If from '@app/components/If/index';
import { selectShowLoader, selectPlayingSong } from '../ItunesAppContainer/selectors';
import SongThumbNail from '../ItunesAppContainer/SongThumbNail/index';
import { itunesAppContainerCreators } from '../ItunesAppContainer/reducer';

const CustomCard = styled(Card)`
  && {
    cursor: pointer;
    margin-top: 10px;
  }
`;

export function SongList({ songs = [], showLoader, dispatchSelectedSong, playingSong ,dispatchPlayingSong}) {
  return (
    <div data-testid="song-list">
      <div className="site-card-wrapper">
        <div className="row">
          {songs?.map((songItem, i) => (
            <If condition={songItem?.trackName} key={i}>
              <div className="col-lg-3 col-sm-12">
                <If condition={showLoader}>
                  <CustomCard>
                    <Skeleton paragraph={{ rows: 6 }} loading={showLoader} active></Skeleton>
                  </CustomCard>
                </If>
                <If condition={!showLoader}>
                  <CustomCard
                    onClick={() => dispatchSelectedSong(songItem)}
                    cover={<SongThumbNail dispatchPlayingSong={dispatchPlayingSong} playingSong={playingSong} song={songItem} />}
                    title={songItem?.trackName}
                    bordered={true}
                  >
                    <span> {songItem?.trackName}</span>
                  </CustomCard>
                </If>
              </div>
            </If>
          ))}
        </div>
      </div>
    </div>
  );
}

const mapStateToProps = createStructuredSelector({
  showLoader: selectShowLoader(),
  playingSong: selectPlayingSong(),
});

const mapDispatchToProps = dispatch => {
  const { setSelectedSong, setPlayingSong } = itunesAppContainerCreators;
  return {
    dispatchSelectedSong: song => dispatch(setSelectedSong(song)),
    dispatchPlayingSong: song => dispatch(setPlayingSong(song))
  };
};

SongList.propTypes = { songs: PropTypes.array, showLoader: PropTypes.bool, dispatchSelectedSong: PropTypes.func };

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SongList);
