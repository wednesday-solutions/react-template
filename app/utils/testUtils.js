import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import configureStore from '@app/configureStore';
import { DEFAULT_LOCALE, translationMessages } from '@app/i18n';
import ConnectedLanguageProvider from '@containers/LanguageProvider';
import { IntlGlobalProvider } from '@components/IntlGlobalProvider';
import { create } from 'react-test-renderer';

export const renderWrappedComponent = component =>
  create(
    <Provider store={configureStore({}, browserHistory).store}>
      <IntlProvider locale="en">{component}</IntlProvider>
    </Provider>
  );

export const renderWithIntl = children =>
  render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );

export const getComponentStyles = (Component, props = {}) => {
  renderWithIntl(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);
  // eslint-disable-next-line no-underscore-dangle
  return window.getComputedStyle(componentRoots[0])._values;
};

export const renderProvider = children => {
  const store = configureStore({}, browserHistory).store;
  return render(
    <Provider store={store}>
      <ConnectedLanguageProvider messages={translationMessages}>
        <ThemeProvider
          theme={{
            main: 'violet'
          }}
        >
          {children}
        </ThemeProvider>
      </ConnectedLanguageProvider>
    </Provider>
  );
};
export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});
