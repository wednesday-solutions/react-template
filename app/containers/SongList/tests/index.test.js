/**
 *
 * Tests for SongList
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import SongList from '../index';

describe('<SongList />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<SongList />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 SongList component', () => {
    const { getAllByTestId } = renderWithIntl(<SongList />);
    expect(getAllByTestId('song-list').length).toBe(1);
  });
});
