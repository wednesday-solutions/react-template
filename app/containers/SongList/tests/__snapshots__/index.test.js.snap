// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SongList /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      data-testid="song-list"
    >
      <div
        class="site-card-wrapper"
      >
        <div
          class="ant-row"
          style="margin-left: -5px; margin-right: -5px;"
        />
      </div>
    </div>
  </div>
</body>
`;
